src/util/backend.js

```diff
@@ -284,34 +279,39 @@ class MaxValueTracker {
       throw new Error('entry "' + JSON.stringify(entry) + '" must have a value')
     }
 
-    const last = this.highest
-    const lastValue = new BigNumber(last.value)
-
-    if (lastValue.lt(entry.value)) {
-      this.highest = entry
-
-      if (this.store) {
-        this.writeQueue = this.writeQueue
-          .then(() => {
-            return this.store.put(this.key + ':mvt:maximum', JSON.stringify({
-              value: entry.value,
-              data: entry.data
-            }))
-          })
-
-        await this.writeQueue
-      }
-
-      return last
-    }
-
-    return entry
+    // By wrapping .getMax() and .put() in a Promise and queue it
+    // into writeQueue we ensure that consecutive calls to setIfMax do not
+    // interrupt a previous call
+    let readWritePromise
+    writeQueue = writeQueue.then(() => {
```

Comment:
this callback can be an async function so you can use await inside of it

Reply:
Changed the callback in the next line to use async / await